.cirrus.yml

```diff
@@ -44,10 +44,21 @@ env:  # Global defaults
 # - small: For an x86_64 machine, recommended to have 2 CPUs and 8 GB of memory.
 # - medium: For an x86_64 machine, recommended to have 4 CPUs and 16 GB of memory.
 # - arm64: For an aarch64 machine, recommended to have 2 CPUs and 8 GB of memory.
+#
+# When a contributor maintains a fork of the repo, any pull request they make
+# to their own fork, or to the main repository, will trigger two CI runs:
+# one for the branch push and one for the pull request.
+# This can be avoided by setting NO_BRANCH=1 as a custom env variable in Cirrus.
 
 # https://cirrus-ci.org/guide/tips-and-tricks/#sharing-configuration-between-tasks
 filter_template: &FILTER_TEMPLATE
-  skip: $CIRRUS_REPO_FULL_NAME == "bitcoin-core/gui" && $CIRRUS_PR == ""  # No need to run on the read-only mirror, unless it is a PR. https://cirrus-ci.org/guide/writing-tasks/#conditional-task-execution
+  # No need to run on the read-only mirror, unless it is a PR.
+  # https://cirrus-ci.org/guide/writing-tasks/#conditional-task-execution
+  skip: $CIRRUS_REPO_FULL_NAME == "bitcoin-core/gui" && $CIRRUS_PR == ""
+  # We use only_if rather than skip when NO_BRANCH is set. Otherwise when a
+  # contributor opens a pull request, they will initially see all jobs marked
+  # as skipped.
```

Comment:
What I remember from a while ago, but I haven't retested, is that under a pull request all the checks initially are marked as skipped. But then they show as either finished or failed as the results come in. It was rather flaky.

I could try again and make some screenshots...

Reply:
Glad I checked this, because the glitched described here is no longer happening:
https://github.com/bitcoin/bitcoin/pull/29274#issuecomment-1967030916

Reply:
> I could try again and make some screenshots...

Thanks, I wouldn't make an extra effort to understand the broken behavior you were trying to avoid. But maybe consider  dropping the word "initially" because I think the comment would make more sense without it. The idea of not avoiding "skip" directive to avoid tests showing up as "skipped" is fairly clear. It's just not clear why they tests might initially show up as skipped but then later show up not skipped, if that's what happens.

It might also help if this said what the code does, instead of just what it does not do. The line
`only_if: $NO_BRANCH != "true" || $CIRRUS_PR != ""` is a hard to parse, but I think it means: Run this job if is part of a PR, and also run it on standalone branches, as the project has not set a custom NO_BRANCH variable. Not too important, but I think it could help if comment said before describing why skip was not used.